common/db.go

```diff
@@ -599,15 +606,52 @@ func buildWinningTicketsQuery(sessionIDs []string) string {
 }
 
 func buildSelectOrchsQuery(filter *DBOrchFilter) (string, error) {
-	query := "SELECT serviceURI, ethereumAddr FROM orchestrators WHERE updatedAt >= datetime('now','-1 day')"
-	if filter != nil && filter.MaxPrice != nil {
-		fixedPrice, err := PriceToFixed(filter.MaxPrice)
-		if err != nil {
-			return "", err
+	query := "SELECT serviceURI, ethereumAddr, pricePerPixel, activationRound, deactivationRound FROM orchestrators "
+	fil, err := buildFilterOrchsQuery(filter)
+	if err != nil {
+		return "", err
+	}
+	return query + fil, nil
+}
+
+func buildFilterOrchsQuery(filter *DBOrchFilter) (string, error) {
+	qry := "WHERE updatedAt >= datetime('now','-1 day')"
+	if filter != nil {
+		if filter.MaxPrice != nil {
+			fixedPrice, err := PriceToFixed(filter.MaxPrice)
+			if err != nil {
+				return "", err
+			}
+			qry += " AND pricePerPixel <= " + strconv.FormatInt(fixedPrice, 10)
 		}
-		query = query + " AND pricePerPixel <= " + strconv.FormatInt(fixedPrice, 10)
+
+		if filter.CurrentRound != nil {
+			currentRound := filter.CurrentRound.Int64()
+			qry += fmt.Sprintf(" AND activationRound <= %v AND %v < deactivationRound", currentRound, currentRound)
+		}
+	}
+	return qry, nil
+}
+
+func buildUpsertOrchsQuery(orch *DBOrch) string {
+	qry := "INSERT INTO orchestrators(updatedAt, serviceURI, ethereumAddr, pricePerPixel, activationRound, deactivationRound, createdAt)"
```

Comment:
Dynamically generating the SQL query seems like the most straightforward way to implement the upsert without requiring the `orch` argument to reflect the current values of the corresponding row in the table if it exists and it seems that SQL injection *should* be a non-issue since the only strings are `serviceURI`/`ethereumAddr` and we validate that `serviceURI` is actually an URI and `ethereumAddr` needs to conform to the ETH address format. I don't see a better alternative at the moment, but maybe @j0sh has some thoughts

Reply:
@yondonfu @kyriediculous Looks like SQLite's `CASE` statement can be used to achieve what we want here via a prepared statement. See https://stackoverflow.com/a/4874440 for a usage example.

The query might end up being long, so it wouldn't be terrible to break it across lines. Something like:

```golang
stmt, err := db.Prepare(`
INSERT INTO orchestrators(updatedAt, serviceURI, ethereumAddr, ... )
VALUES (?, ?, ?, ... )
ON CONFLICT(ethereumAddr)  DO UPDATE SET
serviceURI =
  CASE WHEN trim(excluded.serviceURI) == ""
  THEN orchestrators.serviceURI
  ELSE trim(excluded.serviceURI) END,
pricePerPixel =
  CASE WHEN excluded.pricePerPixel == 0 THEN ...
...
`)
```

Since there are a lot of parameters here, I also wonder if binding to [named params](https://www.sqlite.org/c3ref/bind_parameter_name.html) would help readability? Although I haven't looked into how to use those within the go-sqlite3 bindings. Or at some point we could start using a code generator which would cut down on the boilerplate and probably produce a more ergonomic API, but that's not terrible important right now.

BTW this is a nice use case of a the relatively new ON CONFLICT statement - it hadn't been released when we were originally working  on the orchestrator DB stuff.

Reply:
Thanks for the tip @j0sh!

@kyriediculous I think we should try to stick with prepared statements whenever possible - what do you think about using `CASE` here to setup the prepared statement instead of dynamically generating the statement?

> Since there are a lot of parameters here, I also wonder if binding to named params would help readability?

Seems like it should be possible by passing in a [sql.NamedArg](https://golang.org/pkg/database/sql/#Named) but apparently [go-sqlite3](https://github.com/mattn/go-sqlite3/issues/697) only accepts parameter names of the form `:foo` at the moment.

Reply:
using `CASE` sounds good to me, I'll check it out.

Reply:
It does lead to quite a bit of code duplication whenever we use `DBOrchFilter` though, we currently build that part of the query with a helper than is used in both `selectOrchs` and `orchCount`

Reply:
using `CASE` and named args in 670346d